### PR TITLE
[FIX] website_event: save utm with website event

### PR DESCRIPTION
**Pull request by hamzaislam101**
The utm parameters i-e utm_source, utm_campaign, utm_medium, are only set when a sale order is generated against an event and not set if there is no sale against an event registration that is done through a tracked link. This functionality was missing for the event registration part.

### Steps to reproduce:
1. Create an event and copy its website link.
2. Go to the link tracker and create a tracked link with utm values filled.
3. Use this tracked link in incognito preferably to register for the event.
4. Check the attendees of the event and check the marketing utm values.

### Current Behavior:
The marketing values are not filled, because they are not set. The utm values are computed using sale order and if there is no flow to set these values.

### Expected Behavior:
The utm marketing values should be set.

OPW-3222179

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by reth-odoo — COMMENTED** [review]
Indeed, seems like an oversight from efd5401ee1efbdd388d9623d59a96353e93bf21d

I think it would actually be ok to just inherit 'utm.mixin' now. All the fields are already there so the only change would be that we update the model so the fields become inherited, and we get the mixin's default_get which does all the work for the other models.

If runbot doesn't flag it I don't think there should be any issue with that. If it flags it, we could implement it more similarly to the mixin by using default_get, just in case there's other ways in which it doesn't work.

Objection to adding an inherit in stable in this case @tde-banana-odoo ?

**Comment by hamzaislam101** [comment]
@thiryr Should I push this change or are we waiting for approval?
image

**Comment by thiryr** [comment]
Hey :wave: 

arf, I didn't take not of the fact the fields were named differently on 14.0
It means we can't do this because the fields would not line up in the database.

So instead I'd say let's 'copy' the `default_get` method from `utm.mixin` into `event.registration`, setting campaign_id and such based on the context keys in the model directly instead of doing that in the http route handling.

Then stop the forward-port in 15.0, and there just add utm.mixin to `_inherit`. I checked with someone from framework-py recently by chance so I'm confident doing that should work when the fields are named the same :)

Does that sound ok?

**Comment by hamzaislam101** [comment]
Alright, I'll add these and let you know.

**Comment by hamzaislam101** [comment]
@thiryr Can you please check the updated code?

File: addons/event/models/event_registration.py [hunk 1]
```diff
@@ -148,12 +149,53 @@ def _synchronize_partner_values(self, partner, fnames=None):
...
+                    records = Model.search([('name', '=', value)], limit=1)
+                    if not records:
+                        if 'is_website' in records._fields:
+                            records = Model.create({'name': value, 'is_website': True})
+                        else:
+                            records = Model.create({'name': value})
+                    value = records.id
+                if value:
+                    values[field_name] = value
+        return values
```
Comment (thiryr) [inline addons/event/models/event_registration.py:189]: Can we put this in default_get? Would avoid the 'vals or ...' in the create and it's more consistent with the mixin

File: addons/event/models/event_registration.py [hunk 2]
```diff
@@ -148,12 +149,53 @@ def _synchronize_partner_values(self, partner, fnames=None):
...
+            vals['utm_medium_id'] = vals.get('utm_medium_id') or self.get_utm_params(['utm_medium_id']).get('utm_medium_id')
+
         registrations = super(EventRegistration, self).create(vals_list)
         if registrations._check_auto_confirmation():
             registrations.sudo().action_confirm()
 
         return registrations
 
+    def get_utm_params(self, fields):
+        values = {}
```
Comment (thiryr) [inline addons/event/models/event_registration.py:164]: In the default_get, can we add a short comment/docstring to specify it emulates utm.mixin?

**Review by reth-odoo — COMMENTED** [review]
Couple comments

We'll also **need** a short test that calls this, otherwise runbot won't allow us to update the method since it will never fail.

We also have a task for this actually if you'd rather we handle it, up to you

**Comment by hamzaislam101** [comment]
@thiryr Updated with suggestions. I am actually quite not sure about the process for testing values set in cookies. So if it's in your pipeline then I am happy with the test being handled by you or your team.

**Comment by thiryr** [comment]
Alright, thanks for the work so far then. I'll mention this PR when I get to that task, cheers :)

**Pull request closed without merging**